In [1]:
import pandas as pd

In [2]:
df=pd.read_csv('titanic.csv')
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [3]:
data=df[['survived','pclass','who']]
data

survived  pclass    who
0           0       3    man
1           1       1  woman
2           1       3  woman
3           1       1  woman
4           0       3    man
..        ...     ...    ...
886         0       2    man
887         1       1  woman
888         0       3  woman
889         1       1    man
890         0       3    man

[891 rows x 3 columns]

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   survived  891 non-null    int64 
 1   pclass    891 non-null    int64 
 2   who       891 non-null    object
dtypes: int64(2), object(1)
memory usage: 21.0+ KB


In [5]:
data.head()

survived  pclass    who
0         0       3    man
1         1       1  woman
2         1       3  woman
3         1       1  woman
4         0       3    man

In [6]:
#결측치 확인
data.isna().sum()

survived    0
pclass      0
who         0
dtype: int64

In [7]:
import numpy as np
data['survived'].replace(0,np.nan)

0      NaN
1      1.0
2      1.0
3      1.0
4      NaN
      ... 
886    NaN
887    1.0
888    NaN
889    1.0
890    NaN
Name: survived, Length: 891, dtype: float64

In [20]:
#생존자 리스트
data2=data.dropna()
data2

survived  pclass    who
0           0       3    man
1           1       1  woman
2           1       3  woman
3           1       1  woman
4           0       3    man
..        ...     ...    ...
886         0       2    man
887         1       1  woman
888         0       3  woman
889         1       1    man
890         0       3    man

[891 rows x 3 columns]

In [9]:
#생존자 중 어린이 리스트
data_child=data2[data2['who']=='child']
data_child

survived  pclass    who
7           0       3  child
9           1       2  child
10          1       3  child
14          0       3  child
16          0       3  child
..        ...     ...    ...
831         1       2  child
850         0       3  child
852         0       3  child
869         1       3  child
875         1       3  child

[83 rows x 3 columns]

In [10]:
data_child.info()

<class 'pandas.core.frame.DataFrame'>
Index: 83 entries, 7 to 875
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   survived  83 non-null     int64 
 1   pclass    83 non-null     int64 
 2   who       83 non-null     object
dtypes: int64(2), object(1)
memory usage: 2.6+ KB


In [11]:
#생존한 어린이들을 좌석 등급을 기준으로 정렬
data_child.sort_values('pclass')

survived  pclass    who
435         1       1  child
802         1       1  child
689         1       1  child
297         0       1  child
305         1       1  child
..        ...     ...    ...
233         1       3  child
205         0       3  child
184         1       3  child
352         0       3  child
875         1       3  child

[83 rows x 3 columns]

In [12]:
#1등실 어린이
child1=data_child['pclass']==1
data_child[child1]

survived  pclass    who
297         0       1  child
305         1       1  child
435         1       1  child
445         1       1  child
689         1       1  child
802         1       1  child

In [13]:
data_child[child1].value_counts()

survived  pclass  who  
1         1       child    5
0         1       child    1
Name: count, dtype: int64

In [14]:
#2등실 어린이
child2=data_child['pclass']==2
data_child[child2]

survived  pclass    who
9           1       2  child
43          1       2  child
58          1       2  child
78          1       2  child
183         1       2  child
193         1       2  child
237         1       2  child
340         1       2  child
407         1       2  child
446         1       2  child
530         1       2  child
535         1       2  child
549         1       2  child
618         1       2  child
720         1       2  child
750         1       2  child
755         1       2  child
827         1       2  child
831         1       2  child

In [15]:
data_child[child2].value_counts()

survived  pclass  who  
1         2       child    19
Name: count, dtype: int64

In [16]:
#3등실 어린이
child3=data_child['pclass']==3
data_child[child3]

survived  pclass    who
7           0       3  child
10          1       3  child
14          0       3  child
16          0       3  child
22          1       3  child
24          0       3  child
39          1       3  child
50          0       3  child
59          0       3  child
63          0       3  child
111         0       3  child
119         0       3  child
125         1       3  child
147         0       3  child
164         0       3  child
165         1       3  child
171         0       3  child
172         1       3  child
182         0       3  child
184         1       3  child
205         0       3  child
233         1       3  child
261         1       3  child
278         0       3  child
348         1       3  child
352         0       3  child
374         0       3  child
381         1       3  child
386         0       3  child
419         0       3  child
448         1       3  child
469         1       3  child
479         1       3  child
480         0       3  child
489         1       3  child
541         0       3  child
542         0       3  child
634         0       3  child
642         0       3  child
644         1       3  child
683         0       3  child
686         0       3  child
691         1       3  child
731         0       3  child
751         1       3  child
777         1       3  child
780         1       3  child
787         0       3  child
788         1       3  child
803         1       3  child
813         0       3  child
819         0       3  child
824         0       3  child
830         1       3  child
850         0       3  child
852         0       3  child
869         1       3  child
875         1       3  child

In [17]:
data_child[child3].value_counts()

survived  pclass  who  
0         3       child    33
1         3       child    25
Name: count, dtype: int64

In [18]:
#총 탑승자 중 어린이
child=data['who']=='child'
data[child]

survived  pclass    who
7           0       3  child
9           1       2  child
10          1       3  child
14          0       3  child
16          0       3  child
..        ...     ...    ...
831         1       2  child
850         0       3  child
852         0       3  child
869         1       3  child
875         1       3  child

[83 rows x 3 columns]

In [19]:
#총 생존자
p_survived=data['survived']==1
data[p_survived]

survived  pclass    who
1           1       1  woman
2           1       3  woman
3           1       1  woman
8           1       3  woman
9           1       2  child
..        ...     ...    ...
875         1       3  child
879         1       1  woman
880         1       2  woman
887         1       1  woman
889         1       1    man

[342 rows x 3 columns]

총 탑승자 891명 중 어린이는 83명이었고, 그 중 생존한 어린이의 수는 49명입니다.

생존한 어린이 중 1등실 어린이 5명, 2등실 어린이 19명, 3등실 어린이 25명으로

낮은 등실의 어린이일수록 더 많이 살아남은 것을 알 수 있었습니다.

전체 생존자 342명 중 어린이의 수가 83명으로 상당히 낮은 비율을 차지하는 것을 알 수 있었습니다.

당시 시대상황과 견주어 보았을 때,
